agent/agent.go

```diff
@@ -228,7 +242,24 @@ func (r *Agent) setupExtractor(ctx context.Context, sr recipe.PluginRecipe, str
 	}, nil
 }
 
-func (r *Agent) setupProcessor(ctx context.Context, pr recipe.PluginRecipe, str *stream) error {
+func (r *Agent) setupProcessor(ctx context.Context, pr recipe.PluginRecipe, str *stream, recipeName string) error {
+	var (
+		pluginInfo = PluginInfo{
+			RecipeName: recipeName,
+			PluginName: pr.Name,
+			PluginType: "processor",
+			StartTime:  time.Now(),
+		}
+		err error
+	)
+
+	defer func() {
+		pluginInfo.Success = err == nil
+		for _, monitor := range r.monitor {
+			monitor.RecordPlugin(ctx, pluginInfo)
```

Comment:
This won't work. We would end up recording the metrics before even the processor has had a chance to run. Instead we should look to implement an `OtelProcessorMiddleware` that can emit metrics on each call of `Process`.

Reply:
updated